File: nodejs/src/testConnection.js
import {getBusinessCollection, closeConnection} from './config.js';

async function runAggregation() {
    let coll;
    try {
        coll = await getBusinessCollection();
        console.log('Successfully connected to Atlas');


        const pipeline = [
            {$group: {_id: '$city', count: { $sum: 1 }}}, {$sort: { count: -1 }}, {$limit: 5}
        ];

        const topTen = await coll.aggregate(pipeline).toArray();
        console.log('Top 5 Cities by Business Count:');
        topTen.forEach(({ _id, count }) =>
            console.log(`  ${_id}: ${count}`)
        );

} catch (err) {
        console.error('Error:', err);
    process.exitCode = 1;
    } finally {
        await closeConnection();
        console.log('Connection closed');
    }
}

runAggregation();

File: python/tasks/testConnection.py
import sys
from mongo_client import get_client
from config import DATABASE_NAME

def run():
    client = None
    try:
        client = get_client()
        print("Successfully connected to Atlas")

        db = client[DATABASE_NAME]
        coll = db.business

        pipeline = [
            {"$group": {"_id": "$city", "count": {"$sum": 1}}},
            {"$sort": {"count": -1}},
            {"$limit": 5}
        ]
        top_five = list(coll.aggregate(pipeline))
        print("Top 5 Cities by Business Count:")
        for doc in top_five:
            print(f"  {doc['_id']}: {doc['count']}")

    except Exception as err:
        print("Error:", err)
        sys.exit(1)

    finally:
        if client:
            client.close()
            print("\nConnection closed")

if __name__ == "__main__":
    run()
